src/main/java/at/hannibal2/skyhanni/features/inventory/FannCost.kt

```diff
@@ -0,0 +1,206 @@
+    )
+    private val dailyExpPattern by patternGroup.pattern(
+        "exp.daily",
+        """EXP Per Day: ([\d,]+)(?: \(\+\d{1,2}(\.\d{1,2})?%\))?""",
+    )
+    private val durationPattern by patternGroup.pattern(
+        "training.duration.pattern",
+        """Will take: (?<day>\d+)d (?<hr>\d{1,2})h (?<min>\d{1,2})m (?<sec>\d{1,2})s""",
+    )
+    private val coinsPattern by patternGroup.pattern(
+        "coin",
+        """(\d{1,3}(?:,\d{3})*(?:\.\d+)?|\d+\.?\d*) Coins(?: \([1-5]% off\))?""",
+    )
+    private val bitsPattern by patternGroup.pattern(
+        "bits",
+        """(\d{1,3}(?:,\d{3})*(?:\.\d+)?|\d+\.?\d*) Bits""",
+    )
+    private val desiredLevelPatter by patternGroup.pattern(
+        "slot24.name.level",
+        "Desired Level: (200|1?[0-9]?[0-9])",
+    )
+    private val userInputPattern by patternGroup.pattern(
+        "slot24.name.input",
+        "User Input",
+    )
+    private val trainingTypePattern by patternGroup.pattern(
+        "training.type",
+        "Type: (Free|Light|Moderate|Expert|Ultra|Turbo!)",
+    )
+
+    @SubscribeEvent
+    fun onFannAnvilTooltip(event: LorenzToolTipEvent) {
+        if (!trainingSlotInventoryPattern.matches(InventoryUtils.openInventoryName())) return
+        if (!anvilPattern.matches(event.itemStack.displayName)) return
+        val tooltip = event.toolTip
+
+        val trainingType = tooltip.getTrainingType() ?: return
+
+        when (trainingMode) {
+            TrainingMode.DAY_COUNT -> {
+                if (trainingType == TrainingType.FREE) return
+                if (!showCoins || !showBits) return
+
+                val totalExp = tooltip.getExpEarned() ?: return
+                val coinPerExp = tooltip.getCoins() / totalExp
+                val xpPerBit = totalExp / tooltip.getBits()
+
+                tooltip.insertLineAfter(coinsPattern, "   §6Coins/XP: ${coinPerExp.roundTo(2)}")
+                tooltip.insertLineAfter(bitsPattern, "   §6XP/Bit: ${xpPerBit.roundTo(2)}")
+
+            }
+
+            TrainingMode.UNTIL_LEVEL -> {
+                if (trainingType == TrainingType.FREE) return
+                if (!showCoins || !showBits) return
+
+                val dailyExp = tooltip.getDailyExp() ?: return
+                val duration = tooltip.getDuration() ?: return
+                val totalExp = dailyExp * duration
+                val coinPerExp = tooltip.getCoins() / totalExp
+                val xpPerBit = totalExp / tooltip.getBits()
+                tooltip.insertLineAfter(coinsPattern, "   §6Coins/XP: ${coinPerExp.roundTo(2)}")
+                tooltip.insertLineAfter(bitsPattern, "   §6XP/Bit: ${xpPerBit.roundTo(2)}")
+
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryUpdate(event: InventoryUpdatedEvent) {
+        if (!trainingSlotInventoryPattern.matches(event.inventoryName.removeColor())) return
+        val slot24 = event.inventoryItems[24]
```

Comment:
instead of assigning a variable then checking if it is null, just check if slot 24 is null wiht an elvis operator to then return from the function